pkg/channel/message_receiver.go

```diff
@@ -211,10 +218,10 @@ func (r *MessageReceiver) fromHTTPHeaders(headers http.Header) map[string]string
 
 // ParseChannel converts the channel's hostname into a channel
 // reference.
-func ParseChannel(host string) (channel.ChannelReference, error) {
-	chunks := strings.Split(host, ".")
+func ParseChannel(host url.URL) (channel.ChannelReference, error) {
+	chunks := strings.Split(host.Path, ".")
```

Comment:
this is pulling the path, not the host. is that an issue?

Reply:
It matches the eventing repo: https://github.com/knative/eventing/blob/702be2256abf1be34390b7eb4ea63f600d46ba8d/pkg/channel/event_receiver.go#L185

Reply:
@shefaliv and I looked into this because it was odd to me too. It works because of a quirk of [`url.Parse`](https://golang.org/pkg/net/url/#Parse). If a URI has no scheme, url.Parse assumes that's a relative uri and puts the string in Path. So this doesn't actually change any behavior even though it looks wrong. FWIW a unit test in eventing failed when we tested switching it to use Host, so there is coverage for this.

What we really want here is url.ParseRequestURI which assumes a scheme-less uri is absolute and puts the string in Host. But we'll change that in a followup to eventing and eventing-contrib.